pkg/cmd/controller/cmd.go

```diff
@@ -11,10 +14,20 @@ import (
 )
 
 func NewController() *cobra.Command {
+	var featureGates map[string]bool
+
 	cmd := controllercmd.
-		NewControllerCommandConfig("service-ca-controller", version.Get(), controller.StartServiceCAControllers, clock.RealClock{}).
+		NewControllerCommandConfig("service-ca-controller", version.Get(), func(ctx context.Context, controllerContext *controllercmd.ControllerContext) error {
+			shortCertRotationEnabled := false
+			if val, ok := featureGates["ShortCertRotation"]; ok && val {
+				shortCertRotationEnabled = true
+			}
+			return controller.StartServiceCAControllers(ctx, controllerContext, shortCertRotationEnabled)
```

Comment:
Nit: This logic is unnecessary since it matches the behavior of `shortCertRotationEnabled := featureGates["ShortCertRotation"]`.

```suggestion
			return controller.StartServiceCAControllers(ctx, controllerContext, featureGates["ShortCertRotation"])
```

Reply:
Oh indeed, updated